skopt/dummy_opt.py

```diff
@@ -52,20 +67,37 @@ def dummy_minimize(func, dimensions, n_calls=1000, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=n_calls, random_state=rng)
-
-    init_y = func(X[0])
+    if x0 is None:
+        x0 = []
+    if type(x0) is not list:
+        x0 = list(x0)
+    if len(x0) > 0 and type(x0[0]) is not list:
+        x0 = [x0]
+    n_random_starts = n_calls
+    if y0 is None:
+        y0 = [func(x) for x in x0]
+        n_random_starts -= len(y0)
+    if type(y0) is not list:
```

Comment:
We had decided that the `rvs` from Space objects should return a list of lists over here (https://github.com/scikit-optimize/scikit-optimize/issues/86) . TLDR; it used to return a numpy array of dtype object when the elements of the lists are of different types (float | int)

So for this PR to make things easier, I would suggest to just support list of lists.

For y, I  have not strong opinion and I am ok both ways (whether to support lists, iterables either in this PR or outside)